engine/src/main/java/org/terasology/config/SystemConfig.java

```diff
@@ -16,99 +16,63 @@
-        this.maxThreads = maxThreads;
-    }
-
-    public int getMaxSecondsBetweenSaves() {
-        return maxSecondsBetweenSaves;
-    }
-
-    public void setMaxSecondsBetweenSaves(int maxSecondsBetweenSaves) {
-        this.maxSecondsBetweenSaves = maxSecondsBetweenSaves;
-    }
-
-    public int getMaxUnloadedChunksPercentageTillSave() {
-        return maxUnloadedChunksPercentageTillSave;
-    }
-
-    public void setMaxUnloadedChunksPercentageTillSave(int maxUnloadedChunksPercentageTillSave) {
-        this.maxUnloadedChunksPercentageTillSave = maxUnloadedChunksPercentageTillSave;
-    }
-
-    public boolean isDebugEnabled() {
-        return debugEnabled;
-    }
-
-    public void setDebugEnabled(boolean debugEnabled) {
-        this.debugEnabled = debugEnabled;
-    }
-
-    public boolean isMonitoringEnabled() {
-        return monitoringEnabled;
-    }
-
-    public void setMonitoringEnabled(boolean monitoringEnabled) {
-        this.monitoringEnabled = monitoringEnabled;
-    }
-
-    public boolean isWriteSaveGamesEnabled() {
-        String property = System.getProperty(SAVED_GAMES_ENABLED_PROPERTY);
-        if (property != null) {
-            return Boolean.parseBoolean(property);
-        }
-        return writeSaveGamesEnabled;
-    }
-
-    public void setWriteSaveGamesEnabled(boolean writeSaveGamesEnabled) {
-        this.writeSaveGamesEnabled = writeSaveGamesEnabled;
-    }
-
-    public long getChunkGenerationFailTimeoutInMs() {
-        return chunkGenerationFailTimeoutInMs;
-    }
-
-    public void setChunkGenerationFailTimeoutInMs(long chunkGenerationFailTimeoutInMs) {
-        this.chunkGenerationFailTimeoutInMs = chunkGenerationFailTimeoutInMs;
-    }
-
-    public Locale getLocale() {
-        if (locale == null) {
-            setLocale(Locale.getDefault(Category.DISPLAY));
-        }
-        return Locale.forLanguageTag(locale);
-    }
-
-    public void setLocale(Locale locale) {
-        this.locale = locale.toLanguageTag();
-    }
+    public final Setting<Long> dayNightLengthInMs = setting(
+            type(Long.class),
+            defaultValue(1800000L),
+            constraint(new NumberRangeConstraint<>(0L, Long.MAX_VALUE, false, false))
+    );
+    public final Setting<Integer> maxThreads = setting(
+            type(Integer.class),
+            defaultValue(Runtime.getRuntime().availableProcessors() - 1),
+            constraint(new NumberRangeConstraint<>(0, Integer.MAX_VALUE, false, false))
+    );
+    public final Setting<Integer> maxSecondsBetweenSaves = setting(
+            type(Integer.class),
+            defaultValue(60),
+            constraint(new NumberRangeConstraint<>(0, Integer.MAX_VALUE, false, false))
+    );
+    public final Setting<Integer> maxUnloadedChunksPercentageTillSave = setting(
+            type(Integer.class),
+            defaultValue(40),
+            constraint(new NumberRangeConstraint<>(0, Integer.MAX_VALUE, false, false))
```

Comment:
If this is about percentage, shouldn't the range constraint be [0...100]?
```suggestion
            constraint(new NumberRangeConstraint<>(0, 100, false, false))
```